### PR TITLE
882 lm herbert

### LINKED ISSUE #882: Add L-M from HORACE
Related to #492 

The minimizer in Horace can be found here:
https://github.com/pace-neutrons/Herbert/blob/master/herbert_core/applications/multifit/%40mfclass/private/multifit_lsqr.m 

This should be included in the available minimizers for FitBenchmarking

### PR DESCRIPTION
**Pull request by tyronerees**
#### Description of Work

Fixes #882 

#### Testing Instructions

1. Run on a matlab-enabled machine
2. Check that Horace/lm-lsqr works as expected
3. Check that the tests are correct **The regression test needs updating**
4. Check that the docs make sense and are complete.

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?

### CODE REVIEW

File: examples/all_softwares.ini [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
-          ralfit
-          scipy
-          scipy_ls
-          scipy_go
+[FITTING]
+software: bumps
+          dfo
+          gradient_free
+          gsl
+	  horace
```
Comment (AndrewLister-STFC) [inline examples/all_softwares.ini:6]: Mind if we change this tab to the equivalent in spaces?
```suggestion
          horace

```

File: fitbenchmarking/controllers/horace_controller.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+from fitbenchmarking.controllers.base_controller import Controller
+from fitbenchmarking.controllers.matlab_mixin import MatlabMixin
+
+eng = matlab.engine.start_matlab()
+
+
+class HoraceController(MatlabMixin, Controller):
+    """
+    Controlller for fit in Herbert
```
Comment (AndrewLister-STFC) [inline fitbenchmarking/controllers/horace_controller.py:16]:
```suggestion
    Controller for fit in Herbert
```

**Review by AndrewLister-STFC — COMMENTED** [review]
Spotted some typos but looks good in general. I'll run the tests and update the regressions before merging

**Comment by tyronerees** [comment]
@AndrewLister-STFC -- do you know how to make this link point to this heading: 

image

Using the title of the heading as the target takes it to the minimizer page of MATLAB.  Setting up a target like this doesn't seem to work either:
image

Any suggestions?

File: docs/source/users/install_instructions/externals.rst [hunk 3]
```diff
@@ -43,6 +52,7 @@ Mantid is used both as fitting software, and to parse data files.
 Instructions on how to install Mantid for a range of systems are available
 at `<https://download.mantidproject.org/>`_.
 
+.. _matlab-install:
```
Comment (AndrewLister-STFC) [inline docs/source/users/install_instructions/externals.rst:55]:
```suggestion
.. _matlab-install:

```

**Comment by AndrewLister-STFC** [comment]
@tyronerees Fixed! It just needs a newline between anchor and heading

File: examples/all_softwares.ini [hunk 4]
```diff
@@ -1,16 +1,18 @@
...
-          scipy_ls
-          scipy_go
+[FITTING]
+software: bumps
+          dfo
+          gradient_free
+          gsl
+          horace
+
+          levmar
```
Comment (AndrewLister-STFC) [inline examples/all_softwares.ini:8]:
```suggestion
          horace
          levmar

```

**Review by AndrewLister-STFC — CHANGES_REQUESTED** [review]
Just 1 minor fix that I'll commit then merge

File: examples/all_softwares.ini [hunk 5]
```diff
@@ -1,16 +1,18 @@
...
-          scipy_go
+[FITTING]
+software: bumps
+          dfo
+          gradient_free
+          gsl
+          horace
+          levmar
+
+          mantid
```
Comment (AndrewLister-STFC) [inline examples/all_softwares.ini:9]:
```suggestion
          levmar
          mantid
```

**Review by AndrewLister-STFC — APPROVED** [review]

**Pull request merged**